tools/chapel-py/chapel/__init__.py

```diff
@@ -216,8 +216,9 @@ def match_inner(ast, pat):
 
     return variables if match_inner(ast, pattern) else None
 
-def each_matching(node, pattern):
-    for child in preorder(node):
+def each_matching(node, pattern, iterator=preorder):
+    iterator_ = iterator(node) if iterator else node
```

Comment:
Since you already have a default value for `iterator`, I think you should just not handle the case where `iterator` is `None` (ie.,no need for the if-expression here)